src/Storage/AuthorizationCodeStorage.php

```diff
@@ -5,120 +5,88 @@
 use OAuth2\OpenID\Storage\AuthorizationCodeInterface;
 
 class AuthorizationCodeStorage implements AuthorizationCodeInterface {
-	const TAXONOMY = 'oidc-authorization-code';
+	const META_KEY_PREFIX = 'oidc';
 
 	private static array $authorization_code_data = array(
 		'code'         => 'string', // authorization code.
 		'client_id'    => 'string', // client identifier.
-		'user_login'   => 'string', // The WordPress user id.
 		'redirect_uri' => 'string', // redirect URI.
 		'expires'      => 'int',    // expires as unix timestamp.
 		'scope'        => 'string', // scope as space-separated string.
 		'id_token'     => 'string', // The OpenID Connect id_token.
 	);
 
-	public function __construct() {
-		// Store the authorization codes in a taxonomy.
-		register_taxonomy( self::TAXONOMY, null );
-		foreach ( self::$authorization_code_data as $key => $type ) {
-			register_term_meta(
-				self::TAXONOMY,
-				$key,
-				array(
-					'type'              => $type,
-					'single'            => true,
-					'sanitize_callback' => array( __CLASS__, 'sanitize_' . $key ),
-				)
-			);
+	public function getAuthorizationCode( $code ) {
+		if ( empty( $code ) ) {
+			return null;
 		}
-	}
-
-	public static function sanitize_string_length( $string, $length ) {
-		return substr( $string, 0, $length );
-	}
-
-	public static function sanitize_code( $code ) {
-		return self::sanitize_string_length( $code, 40 );
-	}
 
-	public static function sanitize_client_id( $client_id ) {
-		return self::sanitize_string_length( $client_id, 200 );
-	}
+		$users = get_users(
+			array(
+				'number'     => 1,
+				'meta_key'   => self::META_KEY_PREFIX . '_code', // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_key
```

Comment:
`WordPress.DB.SlowDBQuery.slow_db_query_meta_key` → this is what worried me when I chose to use a taxonomy.

Reply:
I think since this is not a query in a theme page, where it would run on every page load, we are probably ok? `meta_key` index should help a good deal. I added the `number` arg earlier today so that sql query finishes its scan early. We do have this [issue for cleanup](https://github.com/Automattic/wp-openid-connect-server/issues/34).

We should also try to see how slow this query actually is, as compared to other queries, before wanting to pivot?

Reply:
This is the query that gets run `SELECT wp_users.ID FROM wp_users INNER JOIN wp_usermeta ON ( wp_users.ID = wp_usermeta.user_id ) WHERE 1=1 AND ( ( wp_usermeta.meta_key = 'code' AND wp_usermeta.meta_value = 'val' ) ) ORDER BY user_login ASC LIMIT 0, 1`

Its a `join` but its on primary keys for both table and one of the WHERE column is also indexed, so that would enable faster filteration of data. But it could turn out to be slow with a huge users table, which is why I suggested we should try to get actual query time for it.

And if we can't use meta query on a non-front-end page, where can we? It would seem capability exists but is not practically usable.

Reply:
You could argue that on small installs inefficient queries don't matter too much, hence it's feasible to have the ability to query by that field.

If we just make the code part of the meta key (e.g. `oidc_code_123...def`) then we can efficiently query for it. We could store the expiry in a key `oidc_expiry_123...def` (which would reduce the max code length to 243). In MySQL queries for `oidc_expiry_%` can also use the index to make sure that the query for cleanup can be run fast as well.